pkg/apis/workspace/v1alpha1/workspace_types.go

```diff
@@ -13,48 +13,92 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+// WorkspaceSpec defines the desired state of Workspace
+// +k8s:openapi-gen=true
+type WorkspaceSpec struct {
+	// Whether the workspace should be started or stopped
+	Started bool `json:"started"`
+	// Routing class the defines how the workspace will be exposed to the external network
+	RoutingClass WorkspaceRoutingClass `json:"routingClass,omitempty"`
+	// Workspace Structure defined in the Devfile format syntax.
+	// For more details see the Che 7 documentation: https://www.eclipse.org/che/docs/che-7/making-a-workspace-portable-using-a-devfile/
+	Devfile DevfileSpec `json:"devfile"`
+}
+
+// WorkspaceStatus defines the observed state of Workspace
+// +k8s:openapi-gen=true
+type WorkspaceStatus struct {
+	WorkspaceId string         `json:"workspaceId"`
+	Phase       WorkspacePhase `json:"phase,omitempty"`
+	IdeUrl      string         `json:"ideUrl"`
+	// Conditions represent the latest available observations of an object's state
+	// +listType=map
+	Condition []WorkspaceCondition `json:"condition,omitempty"`
+
+	// TODO: This could potentially be handled via configmap more cleanly
+	AdditionalFields WorkspaceStatusAdditionalFields `json:"additionalFields,omitempty"`
+}
+
+type WorkspaceStatusAdditionalFields struct {
+	Runtime string `json:"org.eclipse.che.workspace/runtime"`
+}
+
+// WorkspaceCondition contains details for the current condition of this workspace.
+type WorkspaceCondition struct {
+	// Type is the type of the condition.
+	Type WorkspaceConditionType `json:"type"`
+	// Phase is the status of the condition.
+	// Can be True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human-readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+type WorkspacePhase string
+
+// Valid workspace Statuses
+const (
+	WorkspaceStatusStarting WorkspacePhase = "Starting"
+	WorkspaceStatusReady    WorkspacePhase = "Ready"
```

Comment:
Previously it was Running, right?
I don't have a strong preferences but just reminder that OpenShift Console depends on Running state and we need inform them after PR is merged.

Reply:
You're right, great catch -- I mixed up the workspace phase and the workspaceRouting phase. Fixed in PR https://github.com/che-incubator/che-workspace-operator/pull/46